backend/loi-service/src/main/java/org/egov/works/util/MDMSUtils.java

```diff
@@ -0,0 +1,105 @@
+package org.egov.works.util;
+
+import org.egov.common.contract.request.RequestInfo;
+import org.egov.mdms.model.MasterDetail;
+import org.egov.mdms.model.MdmsCriteria;
+import org.egov.mdms.model.MdmsCriteriaReq;
+import org.egov.mdms.model.ModuleDetail;
+import org.egov.works.config.LOIConfiguration;
+import org.egov.works.repository.ServiceRequestRepository;
+import org.egov.works.web.models.LetterOfIndent;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.LinkedList;
+import java.util.List;
+
+import static org.egov.works.util.LOIConstants.MDMS_MASTER;
+import static org.egov.works.util.LOIConstants.MDMS_MODULE_NAME;
+
+@Component
+public class MDMSUtils {
+
+
+    private LOIConfiguration config;
+
+    private ServiceRequestRepository serviceRequestRepository;
+
+    @Autowired
+    public MDMSUtils(LOIConfiguration config, ServiceRequestRepository serviceRequestRepository) {
+        this.config = config;
+        this.serviceRequestRepository = serviceRequestRepository;
+    }
+
+    /**
+     * Calls MDMS service to fetch loi master data
+     *
+     * @param request
+     * @return
+     */
+    public Object mDMSCall(LetterOfIndent request) {
```

Comment:
I think there can be a simpler way of making this api call with less code involved. This looks like too much code for a single api call. Discuss with Pintu to find a better solution.

Reply:
Has Pintu implemented this any differently?

Reply:
we are still working on the changes will update the proper change in the next PR as im still not calling mdms from my service